From 9181bd6434d22ecab43861f00dcb8ed56e7fcfe9 Mon Sep 17 00:00:00 2001
From: Rob Crittenden <rcritten@redhat.com>
Date: Wed, 6 Mar 2013 14:28:18 -0500
Subject: [PATCH 1/2] Don't base64-encode the CA cert when uploading it during
 an upgrade.

We want to store the raw value. Tools like ldapsearch will automatically
base64 encode the value because it's binary so we don't want to duplicate
that.

https://fedorahosted.org/freeipa/ticket/3477
---
 ipaserver/install/plugins/upload_cacrt.py | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/ipaserver/install/plugins/upload_cacrt.py b/ipaserver/install/plugins/upload_cacrt.py
index d60247b7c4884d35646cd4cb8ea3435bbcb6c0f5..a82fc36bfaa68be282846aa6ed6426dc4c941132 100644
--- a/ipaserver/install/plugins/upload_cacrt.py
+++ b/ipaserver/install/plugins/upload_cacrt.py
@@ -39,7 +39,6 @@ def execute(self, **options):
         certdb = certs.CertDB(api.env.realm, nssdir=dirname, subject_base=subject_base)
 
         dercert = certdb.get_cert_from_db(certdb.cacert_name, pem=False)
-        cadercert = base64.b64encode(dercert)
 
         updates = {}
         dn = DN(('cn', 'CACert'), ('cn', 'ipa'), ('cn','etc'), api.env.basedn)
@@ -47,7 +46,7 @@ def execute(self, **options):
         cacrt_entry = ['objectclass:nsContainer',
                        'objectclass:pkiCA',
                        'cn:CAcert',
-                       'cACertificate;binary:%s' % cadercert,
+                       'cACertificate;binary:%s' % dercert,
                       ]
         updates[dn] = {'dn': dn, 'default': cacrt_entry}
 
-- 
1.8.1.4

